dev/src/codewind/project/Project.ts

```diff
@@ -468,74 +466,44 @@ export default class Project implements vscode.QuickPickItem {
-                const loadTestPath = path.join(this.localPath.fsPath, "load-test");
-                const timestampPath = path.join(loadTestPath, event.timestamp);
-                let fileName = "";
-                if (this.language === "java") {
-                    fileName = "profiling.hcd";
-                } else if (this.language === "nodejs") {
-                    fileName = "profiling.json";
-                }
-                const profilingOutPath = path.join(timestampPath, fileName);
-                try {
-                    await fs.promises.mkdir(timestampPath, {recursive: true});
-                    const fd = await fs.promises.open(profilingOutPath, "w");
-                    await fd.close();
-                } catch (error) {
-                    Log.e(error);
-                    vscode.window.showErrorMessage(`Could not create file at ${profilingOutPath}`);
-                    return;
-                }
-                const endpoint = ProjectEndpoints.PROFILING.toString().concat(`/${event.timestamp}`);
-                const url = EndpointUtil.resolveProjectEndpoint(this.connection, this.id, endpoint as ProjectEndpoints);
-                let wStream;
-                try {
-                    wStream = fs.createWriteStream(profilingOutPath);
-                } catch (error) {
-                    Log.e(error);
-                    return;
-                }
-                try {
-                    await this.asyncHttpRequest(url, this.connection.url.scheme === "https", wStream);
-                } catch (error) {
-                    Log.e(error);
-                    vscode.window.showErrorMessage(`Error receiving profiling data for ${this.name}`);
-                } finally {
-                    wStream.close();
-                }
-                Log.d(`Saved profiling data to project ${this.name} at ${profilingOutPath}`);
-                break;
-            default:
-                break;
+        if (![ "hcdReady", "profilingReady"].includes(event.status)) {
+            return;
         }
-    }
+        Log.d("Profiling data is ready to be saved to workspace");
+        const loadTestFolder = "load-test";
+        const loadTestPath = path.join(this.localPath.fsPath, loadTestFolder);
+        const timestampPath = path.join(loadTestPath, event.timestamp);
+        let fileName = "";
+        if (this.language.toLowerCase() === ProjectType.Languages.JAVA) {
+            fileName = "profiling.hcd";
+        } else if (this.language.toLowerCase() === ProjectType.Languages.NODE) {
+            fileName = "profiling.json";
+        } else {
+            Log.e(`Project language ${this.language} not supported for profiling`);
+            return;
+        }
+        const profilingOutPath = path.join(timestampPath, fileName);
+        await fs.promises.mkdir(loadTestPath);
+        await fs.promises.mkdir(timestampPath);
+        try {
+            const fd = await fs.promises.open(profilingOutPath, "w");
```

Comment:
is this just a permissions check? if so shouldn't this be done in `httpWriteStreamToFile` ? 

since we just created that directory, it will have the default permissions 755 which means we can write to it already, I'm not sure what this call accomplishes

Reply:
You're right, not needed and removed.